herald/src/main/java/io/heraldprox/herald/sensor/data/TextFile.java

```diff
@@ -22,28 +20,38 @@
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
+import java.util.concurrent.ConcurrentLinkedQueue;
 import java.util.concurrent.Executors;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
 
 public class TextFile implements Resettable {
     private final SensorLogger logger = new ConcreteSensorLogger("Sensor", "Data.TextFile");
+    private final ScheduledExecutorService executorService = Executors.newScheduledThreadPool(1);
     @NonNull
     private final File file;
+    private final ConcurrentLinkedQueue<String> writeBuffer = new ConcurrentLinkedQueue<>();
+    private final AtomicLong writeBufferSize = new AtomicLong();
 
     public TextFile(@NonNull final File file) {
         this.file = file;
         final File folder = file.getParentFile();
-        if (!folder.exists()) {
+        if (folder != null && !folder.exists()) {
             if (!folder.mkdirs()) {
                 logger.fault("Make folder failed (folder={})", folder);
             }
         }
+        executorService.scheduleWithFixedDelay(new Runnable() {
+            @Override
+            public void run() {
+                flush();
+            }
+        }, 30, 30, TimeUnit.SECONDS);
```

Comment:
Is there value in externalising this to a static non-const default setting?

Reply:
Updated to enable configuration if required in constructor, with 30 seconds being the default.